src/ReactiveSocket.cpp

```diff
@@ -151,6 +157,14 @@ bool ReactiveSocket::createResponder(
       handler_->handleFireAndForgetRequest(std::move(frame.data_));
       break;
     }
+    case FrameType::METADATA_PUSH: {
```

Comment:
Since metadata push is always on the connection, I don't think you need this here and it should replace the logic you have in ConnectionAutomaton.